src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  const spiderSize = spider.getBoundingClientRect();
+  const wallRect = wall.getBoundingClientRect();
+
+  if (
+    e.clientX >= wallRect.left &&
+    e.clientX <= wallRect.right &&
+    e.clientY >= wallRect.top &&
+    e.clientY <= wallRect.bottom
+  ) {
+    let targetX = e.clientX - wallRect.left - spiderSize.width / 2 - 10;
+    let targetY = e.clientY - wallRect.top - spiderSize.height / 2 - 10;
```

Comment:
The subtraction of 10 from both `targetX` and `targetY` seems arbitrary. If this offset is intentional for design purposes, it should be documented. Otherwise, consider removing it to ensure the spider centers exactly where clicked.